src/MainWindow.cpp

```diff
@@ -382,6 +395,50 @@ MainWindow::HighScoreCheck(void)
 	}
 }
 
+void
+MainWindow::AchievementCheck(void)
+{
+	uint16 seconds = fTimerView->GetTime();
+	int numAchieved = 0;
+
+	gGamesWon++;
+
+	//General achievement
+	if (gGamesWon > 0 && !gAchievements[3][0]) {
+		gAchievements[3][0] = true;
+		numAchieved++;
+	}
+
+	if (gGamesWon > 4 && !gAchievements[3][1]) {
+		gAchievements[3][1] = true;
+		numAchieved++;
+	}
+
+	//Time based achievements for each difficulty
+	if (gDifficulty < DIFFICULTY_CUSTOM && seconds < 100
+		&& !gAchievements[gDifficulty][0])
+	{
+		gAchievements[gDifficulty][0] = true;
+		numAchieved++;
+	}
+	if (gDifficulty < DIFFICULTY_CUSTOM && seconds < 50
+		&& !gAchievements[gDifficulty][0])
+	{
+		gAchievements[gDifficulty][1] = true;
+		numAchieved++;
+	}
+
+	AchievementWindow *achievementwin = new AchievementWindow();
+	BString title;
+	static BStringFormat format(B_TRANSLATE("{0, plural,"
+		"=1{New achievement unlocked!}"
+		"other{New achievements unlocked!}}"));
+	format.Format(title, numAchieved);
```

Comment:
JFYI, if you don't already know: you can even use "#" to get the numAchieved inserted into the string, like

`other{# new achievements unlocked!}`

Though I see why you may want to keep the window title a bit shorter...

Reply:
Yeah. it's a little on the long side, so I suggest we keep it how it is.